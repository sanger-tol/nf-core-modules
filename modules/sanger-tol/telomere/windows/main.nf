process TELOMERE_WINDOWS {
    tag "${meta.id}"
    label 'process_low'

    conda "bioconda::java-jdk=8.0.112"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/java-jdk:8.0.112--1' :
        'biocontainers/java-jdk:8.0.112--1' }"

    input:
    tuple val(meta), path(telomere)

    output:
    tuple val( meta ), path("*.windows")    , emit: windows
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // WARNING: This module includes the asmstats script as a module binary in
    // ${moduleDir}/resources/usr/bin/telomere.jar. To use this module, you will
    // either have to copy this file to ${projectDir}/bin or set the option
    // nextflow.enable.moduleBinaries = true in your nextflow.config file.
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = "1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    def telomere_jar = task.ext.telomere_jar ?: "telomere.jar"
    def telomere_jvm_params = task.ext.telomere_jvm_params ?: "-Xms1g -Xmx1g"
    def telomere_window_cut = task.ext.telomere_window_cut ?: 99.9

    //${projectDir}/bin/

    """
    java ${telomere_jvm_params} \\
        -cp ${telomere_jar} \\
        FindTelomereWindows $telomere $telomere_window_cut \\
        > ${prefix}.windows

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        telomere: $VERSION
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = "1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    def telomere = task.ext.telomere ?: ''
    """
    touch ${prefix}.windows

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        telomere: $VERSION
    END_VERSIONS
    """

}
