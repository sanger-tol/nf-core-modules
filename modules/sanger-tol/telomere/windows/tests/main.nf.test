nextflow_process {

    name "Test Process TELOMERE_WINDOWS"
    script "../main.nf"
    process "TELOMERE_WINDOWS"

    tag "modules"
    tag "modules_sangertol"
    tag "telomere"
    tag "windows"
    tag "telomere/windows"

    config "./nextflow.config"

    test("Cloeon_dipterum - telomere") {

        when {
            process {
                """
                input[0] = [
                    [ id: "Cloeon_dipterum" ],
                    file(params.modules_testdata_base_path + 'resources/modules/telomere/idFanCani4/idFanCani4.telomere', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success               },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Cloeon_dipterum - telomere - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: "Cloeon_dipterum" ],
                    file(params.modules_testdata_base_path + 'resources/modules/telomere/idFanCani4/idFanCani4.telomere', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success               },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
