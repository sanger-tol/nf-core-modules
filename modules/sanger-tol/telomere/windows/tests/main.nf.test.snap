{
    "Cloeon_dipterum - telomere": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum.windows:md5,3b9105c87ad2c332dd28a868dff9c89e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0677a48a0e2b2426fedfa6d32e885eb8"
                ],
                "versions": [
                    "versions.yml:md5,0677a48a0e2b2426fedfa6d32e885eb8"
                ],
                "windows": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum.windows:md5,3b9105c87ad2c332dd28a868dff9c89e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-13T10:56:10.502346575"
    },
    "Cloeon_dipterum - telomere - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum.windows:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0677a48a0e2b2426fedfa6d32e885eb8"
                ],
                "versions": [
                    "versions.yml:md5,0677a48a0e2b2426fedfa6d32e885eb8"
                ],
                "windows": [
                    [
                        {
                            "id": "Cloeon_dipterum"
                        },
                        "Cloeon_dipterum.windows:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-10-13T10:48:40.477988556"
    }
}